purchase_variant_configurator/views/inherited_purchase_order_views.xml

```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record model="ir.ui.view" id="purchase_order_extended_form_view">
+        <field name="name">purchase.order.extended.form</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="priority" eval="20" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/tree" position="attributes">
+                <!-- force using the form -->
+                <attribute name="editable" />
+            </xpath>
+            <xpath expr="//field[@name='order_line']/form/group[1]" position="before">
+                <group>
+                    <field name="state" invisible="1" />
+                    <!-- HACK: readonly condition is set with an `in` clause instead of a `not in` for allowing on new created lines to be writeable, as
+                             state field is a related one that is not filled until saved -->
+                    <field
+                        name="product_tmpl_id"
+                        attrs="{'invisible': [('display_type', '!=', False)], 'readonly': [('state', 'in', ('purchase', 'to approve','done', 'cancel'))]}"
```

Comment:
According to the migration guide you must change the `attrs` to invisible (the same in the rest of the fields with “_similar_” changes).